flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/DynamicPartitionPruningUtils.java

```diff
@@ -275,6 +400,19 @@ && isSuitableFilter(origProgram.expandLocalRef(origProgram.getCondition()))) {
                 dimSideFactors.hasFilter = true;
             }
             visitDimSide(rel.getInput(0), dimSideFactors);
+        } else if (rel instanceof Join) {
+            Join join = (Join) rel;
+            visitDimSide(join.getLeft(), dimSideFactors);
+            visitDimSide(join.getRight(), dimSideFactors);
+        } else if (rel instanceof BatchPhysicalGroupAggregateBase) {
+            for (RelNode input : rel.getInputs()) {
```

Comment:
there is only one input for BatchPhysicalGroupAggregateBase

Reply:
> there is only one input for BatchPhysicalGroupAggregateBase

Done!